File: dsa/find-duplicates-in-an-array-gfg.cpp
// link : https://www.geeksforgeeks.org/problems/find-duplicates-in-an-array/1
// type : EASY
// concept : Array, Hashing,
/*
  notes:
	vector<int> v(n) => creates a vector of size n with default values as 0.
        v.size() => gives the size of the array.
*/


//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution{
  public:
    vector<int> duplicates(long long arr[], int n) {
        // code here
        vector<int> v(n);
        for(int i=0;i<n;i++){
            v[arr[i]]++;
        }
        vector<int>ans;
        for(int i=0;i<n;i++){
            if(v[i]>1)
            ans.push_back(i);
        }
        if(ans.size()==0){
            ans.push_back(-1);
            return ans;
        }
        return ans;
    }
};


//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t-- > 0) {
        int n;
        cin >> n;
        long long a[n];
        for (int i = 0; i < n; i++) cin >> a[i];
        Solution obj;
        vector<int> ans = obj.duplicates(a, n);
        for (int i : ans) cout << i << ' ';
        cout << endl;
    }
    return 0;
}

// } Driver Code Ends

File: dsa/find-duplicates-in-an-array-gfg.py
# link : https://www.geeksforgeeks.org/problems/find-duplicates-in-an-array/1
# type : EASY
# concept : Array, Hashing
"""
notes :
v = [0]*n => creates a list with size n and init with 0
ans.append(i) => to add the element i to the ans list

"""


class Solution:
    def duplicates(self, arr, n):
        v = [0]*n
        ans = []
        for a in arr:
            v[a]+=1
        for i in range(0,n):
            if v[i] > 1:
                ans.append(i)
        if len(ans) == 0:
            ans.append(-1)
            return ans
        return ans
    	# code here
    	    


#{ 
 # Driver Code Starts
if(__name__=='__main__'):
    t = int(input())
    for i in range(t):
        n = int(input())
        arr = list(map(int, input().strip().split()))
        res = Solution().duplicates(arr, n)
        for i in res:
            print(i,end=" ")
        print()



# } Driver Code Ends
